File: Python3/801 - 1000/976 Largest Perimeter Triangle.py
# 72ms, 98.08%; 14MB, 5.10%
class Solution:
    def largestPerimeter(self, A: list) -> int:
        A.sort(reverse=1)
        for i in range(2, len(A)):
            if A[i - 2] < A[i - 1] + A[i]: return  A[i - 2] + A[i - 1] + A[i]
        return 0

File: 801 - 1000/976 Largest Perimeter Triangle.cc
// 44ms, 96.74%; 10.6MB, 99.20%
static int x = []() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);
	return 0;
}();

class Solution
{
public:
	int largestPerimeter(vector<int>& A)
	{
		sort(A.rbegin(), A.rend());
		for(int i = 2, size = A.size(); i < size; ++i)
			if(A[i - 2] < A[i - 1] + A[i]) return A[i - 2] + A[i - 1] + A[i];
		return 0;
	}
};
